### PR TITLE
REF/TST: Add more pytest idiom to resample/test_base.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ n/a] xref #17806
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ n/a] whatsnew entry

in this pass:

- alternative approach to #24324
- parametrize base class tests individually to avoid changing fixtures

cc @jreback

### CODE REVIEW

File: pandas/tests/resample/test_base.py [hunk 1]
```diff
@@ -28,234 +34,200 @@ def _create_index(*args, **kwargs):
...
+    frame.iloc[1] = None
+    result = frame.resample('1H').asfreq(fill_value=4.0)
+    new_index = create_index(frame.index[0],
+                             frame.index[-1], freq='1H')
+    expected = frame.reindex(new_index, fill_value=4.0)
+    assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    '_index_factory,_series_name,_index_start,_index_end',
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: this is slightly magical

**Review by jreback — CHANGES_REQUESTED** [review]
need to look some more

File: pandas/tests/resample/test_base.py [hunk 2]
```diff
@@ -19,6 +17,22 @@
     assert_almost_equal, assert_frame_equal, assert_index_equal,
     assert_series_equal)
 
+# tuples of '_index_factory,_series_name,_index_start,_index_end'
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: can you give a little more expl here on what this actuall does

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. on master we have 499 tests and 139 skipped, here 497 ad 9 skipped. so I think a couple got removed?

**Comment by simonjayhawkins** [comment]
@jreback 

> on master we have 499 tests and 139 skipped, here 497 ad 9 skipped. so I think a couple got removed?

for `TestDatetimeIndex::test_resample_empty_dtypes`

```
$ pytest pandas/tests/resample/test_base.py::TestDatetimeIndex::test_resample_empty_dtypes
============================= test session starts =============================
platform win32 -- Python 3.6.6, pytest-3.7.3, py-1.5.4, pluggy-0.7.1
rootdir: C:\Users\simon\OneDrive\code\pandas-simonjayhawkins, inifile: setup.cfg
plugins: xdist-1.23.0, forked-0.2, cov-2.5.1, hypothesis-3.59.1
collected 192 items

pandas\tests\resample\test_base.py ..................................... [ 19%]
........................................................................ [ 56%]
........................................................................ [ 94%]
...........                                                              [100%]

========================= 192 passed in 1.55 seconds ==========================
```

similarly for `TestPeriodIndex`

```
$ pytest pandas/tests/resample/test_base.py::TestPeriodIndex::test_resample_empty_dtypes
============================= test session starts =============================
platform win32 -- Python 3.6.6, pytest-3.7.3, py-1.5.4, pluggy-0.7.1
rootdir: C:\Users\simon\OneDrive\code\pandas-simonjayhawkins, inifile: setup.cfg
plugins: xdist-1.23.0, forked-0.2, cov-2.5.1, hypothesis-3.59.1
collected 192 items

pandas\tests\resample\test_base.py ..................................... [ 19%]
........................................................................ [ 56%]
........................................................................ [ 94%]
...........                                                              [100%]

========================= 192 passed in 1.10 seconds ==========================
```

and the same for `TestTimedeltaIndex`

but because `test_resample_empty_dtypes` is paramatrized using a generator it was misbehaving..

```
$ pytest pandas/tests/resample/test_base.py -k test_resample_empty_dtypes
============================= test session starts =============================
platform win32 -- Python 3.6.6, pytest-3.7.3, py-1.5.4, pluggy-0.7.1
rootdir: C:\Users\simon\OneDrive\code\pandas-simonjayhawkins, inifile: setup.cfg
plugins: xdist-1.23.0, forked-0.2, cov-2.5.1, hypothesis-3.59.1
collected 638 items / 318 deselected

pandas\tests\resample\test_base.py ..................................... [ 11%]
........................................................................ [ 34%]
........................................................................ [ 56%]
...........sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 79%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss      [100%]

=========== 192 passed, 128 skipped, 318 deselected in 1.43 seconds ===========

```

so that account for 128 of the 130 reduction in skipped tests. the other two are from the overridden tests...

https://github.com/pandas-dev/pandas/blob/1cd077a3077352d63c1226d554db1e46cb9d2c1a/pandas/tests/resample/test_base.py#L231-L237

the number of passed tests is reduced by two since `test_raises_on_non_datetimelike_index` is not dependent on index type so is now only run once instead of 3 times.

https://github.com/pandas-dev/pandas/blob/1cd077a3077352d63c1226d554db1e46cb9d2c1a/pandas/tests/resample/test_base.py#L71-L74

**Comment by jreback** [comment]
```
_____________ ERROR collecting pandas/tests/resample/test_base.py _____________
2018-12-23T18:59:10.6286537Z C:\Miniconda\envs\pandas-dev\lib\site-packages\_pytest\python.py:450: in _importtestmodule
2018-12-23T18:59:10.6287402Z     mod = self.fspath.pyimport(ensuresyspath=importmode)
2018-12-23T18:59:10.6287666Z C:\Miniconda\envs\pandas-dev\lib\site-packages\py\_path\local.py:668: in pyimport
2018-12-23T18:59:10.6287846Z     __import__(modname)
2018-12-23T18:59:10.6288006Z E     File "D:\a\1\s\pandas\tests\resample\test_base.py", line 21
2018-12-23T18:59:10.6288224Z E   SyntaxError: Non-ASCII character '\xe2' in file D:\a\1\s\pandas\tests\resample\test_base.py on line 21, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
2018-12-23T18:59:10.629434
```

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**